.github/workflows/bean-instantiations.yml

```diff
@@ -0,0 +1,293 @@
+          STARTUP_SEARCH=$(echo "$STARTUP_LOG_PATTERN" | sed 's/ \\(.*$//')
+
+          LINE=$(grep -E "$STARTUP_SEARCH" $LOG_FILE) \
+            || { echo "❌ No startup metrics line found"; cat $LOG_FILE; exit 1; }
+
+          if [[ "$LINE" =~ $STARTUP_LOG_PATTERN ]]; then
+            INSTANTIATED_BEANS=${BASH_REMATCH[1]}
+            LONGEST_CHAIN_LENGTH=${BASH_REMATCH[2]}
+          else
+            echo "❌ Failed to parse startup metrics from: $LINE"
+            exit 1
+          fi
+
+          echo "• Number of instantiated beans = $INSTANTIATED_BEANS"
+          echo "• Longest dependency chain length = $LONGEST_CHAIN_LENGTH"
+
+          echo "Validating against thresholds: MIN_INSTANTIATED_BEANS=$MIN_INSTANTIATED_BEANS (expected ≥ $MIN_INSTANTIATED_BEANS), MAX_INSTANTIATED_BEANS=$MAX_INSTANTIATED_BEANS (expected ≤ $MAX_INSTANTIATED_BEANS), MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH=$MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH (expected ≤ $MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH)"
+
+          STARTUP_VALIDATION_FAILED=false
+
+          if (( INSTANTIATED_BEANS < MIN_INSTANTIATED_BEANS )); then
+            echo "❌ $INSTANTIATED_BEANS < $MIN_INSTANTIATED_BEANS beans. Something seems to be wrong, as usually more beans are instantiated."
+            STARTUP_VALIDATION_FAILED=true
+          fi
+
+          if (( INSTANTIATED_BEANS > MAX_INSTANTIATED_BEANS )); then
+            echo "❌ $INSTANTIATED_BEANS > $MAX_INSTANTIATED_BEANS threshold"
+            STARTUP_VALIDATION_FAILED=true
+          fi
+
+          if (( LONGEST_CHAIN_LENGTH > MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH )); then
+            echo "❌ Longest dependency chain length $LONGEST_CHAIN_LENGTH > $MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH threshold"
+            STARTUP_VALIDATION_FAILED=true
+          fi
+
+          if [ "$STARTUP_VALIDATION_FAILED" = false ]; then
+            echo "✅ Final values → Beans: $INSTANTIATED_BEANS; Longest dependency chain length: $LONGEST_CHAIN_LENGTH"
+            echo "✅ Startup bean instantiation metrics within expected ranges"
+          fi
+
+          # Export variables for next step
+          echo "LONGEST_CHAIN_LENGTH=$LONGEST_CHAIN_LENGTH" >> $GITHUB_ENV
+          echo "STARTUP_VALIDATION_FAILED=$STARTUP_VALIDATION_FAILED" >> $GITHUB_ENV
+
+      - name: Check for startup dependency chains exceeding threshold
+        run: |
+          STARTUP_DEEP_CHAIN_PATTERN="Startup long bean instantiation chain"
+          LOG_FILE="app.log"
+
+          if (( LONGEST_CHAIN_LENGTH > MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH )); then
+            echo "❌ Startup dependency chain length $LONGEST_CHAIN_LENGTH > $MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH threshold"
+            echo "Chains exceeding threshold:"
+            grep -E "$STARTUP_DEEP_CHAIN_PATTERN" $LOG_FILE || echo "  (No detailed chains found in log)"
+            echo "🔧 These chains violate the threshold MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH. Please refactor them to break these chains."
```

Comment:
Can we link some sources on best practices and ways to resolve them? Or is that not feasible / a bad idea? Are there known tools to analyze the chains?